tests/suites/helpers.function

```diff
@@ -149,7 +149,7 @@ typedef enum
     do                                                            \
     {                                                             \
         TEST_ASSERT( ( pointer ) == NULL );                       \
-        if( ( length ) != 0 )                                     \
+        if( ( length ) != 0 && ( length ) < SIZE_MAX )             \
```

Comment:
I don't understand this change. Why would `ASSERT_ALLOC` to transform a request to allocate `SIZE_MAX` bytes into successfully doing nothing? Why would `ASSERT_ALLOC` care about a request to allocate `SIZE_MAX` bytes? Why wouldn't a request to allocate `SIZE_MAX` bytes a potential bug of the caller?

Reply:
I've described the issue above. The issue is actually in `generate_random()`. I could have put a bounds check there, and in hindsight, at the point we receive IO it should always be checked anyway, but I put it before each `calloc`, which is more conservative, and because I was concerned we may get more warnings.